src/app/core/entity/schema/entity-schema.service.ts

```diff
@@ -140,12 +150,24 @@ export class EntitySchemaService {
     const data = {};
 
     for (const key of schema.keys()) {
+      let value = entity[key];
       const schemaField: EntitySchemaField = schema.get(key);
 
-      if (entity[key] !== undefined) {
-        data[key] = this.getDatatypeOrDefault(
-          schemaField.dataType
-        ).transformToDatabaseFormat(entity[key], schemaField, this, entity);
+      if (value === undefined) {
+        if (schemaField.defaultValue !== undefined) {
+          value = schemaField.defaultValue;
```

Comment:
Is this step necessary? When a entity is loaded the value is set and when a entity is saved the value is set.

Reply:
not sure whether I understood your point. This ensures that the defaultValue is passed through the transformation rather than being directly set, to allow for some advanced use cases.

Reply:
My point is that in `transformEntityToDatabaseFormat` the default value is assigned to the attribute if it is undefined. This means no entity can be saved to the database withouth having the default value set. However, `transformDatabaseToEntityFormat` checks again if the value is not set and if a default value is available. Is there a reason why the default value is assigned in both transformations?

Reply:
this is somewhat redundant but a failsafe in case data somehow ends up in the database through an import from other channels - or the data model changes